hack/push-cip.sh

```diff
@@ -0,0 +1,106 @@
+#!/usr/bin/env bash
+
+# Copyright 2020 The Kubernetes Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# About: This script pushes the cip and auditor images to a public repository,
+# defined in workspace_status.sh.
+
+set -o errexit
+set -o nounset
+set -o pipefail
+
+repo_root=$(cd "$(dirname "${BASH_SOURCE[0]}")/.." && pwd -P)
+
+printUsage() {
+    echo "Usage: $0 [--audit | -a]"
+}
+
+# Takes array of images and pushes to their tagged registry location.
+pushImages() {
+    for img in "$@"; do
+        docker push "$img"
+    done
+}
+
+# Builds either auditor or cip container image from repo_root. Also adds
+# tags, given by an array of tags.
+buildImage() {
+    # Add build variant argument.
+    variant=$1
+    cmd="docker build --build-arg variant=$variant "
+    # Concatenate tags.
+    tags="${*:2}"
+    for tag in $tags; do
+        cmd+="-t $tag "
+    done
+    # Specify Dockerfile location.
+    cmd+="$repo_root"
+    # Build the container.
+    echo "Executing: $cmd"
+    $cmd
+}
+
+# Detect which container variant to build.
+build_variant="cip"
+
+if [[ $# == 1 ]]; then
+    if [ "$1" == --audit ] || [ "$1" == -a ]; then
+        build_variant="auditor"
+    else
+        >&2 echo "ERROR: Unknown flag provided."
+        printUsage
+        exit 1
+    fi
+elif [[ $# -gt 1 ]]; then
+    >&2 echo "ERROR: Invalid number of arguments."
+    printUsage
+    exit 1
+fi
+
+# Inject workspace variables.
+source <("${repo_root}"/workspace_status.sh inject)
+
+# Define tags for the auditor.
+# NOTE: Both cip and auditor variants build an auditor image. Although they are named differently,
+# the iamge contents are the exact same. Defaults to cip variant.
+audit_tags=(
+    "${STABLE_IMG_REGISTRY}/${STABLE_IMG_REPOSITORY}/${STABLE_IMG_NAME}-auditor:latest"
+    "${STABLE_IMG_REGISTRY}/${STABLE_IMG_REPOSITORY}/${STABLE_IMG_NAME}-auditor:${STABLE_IMG_TAG}"
+)
+if [ "$build_variant" == auditor ]; then
+    audit_tags=(
+        "${STABLE_TEST_AUDIT_STAGING_IMG_REPOSITORY}/${STABLE_IMG_NAME}-auditor-test:latest"
+        "${STABLE_TEST_AUDIT_STAGING_IMG_REPOSITORY}/${STABLE_IMG_NAME}-auditor-test:${STABLE_IMG_TAG}"
+    )
+fi
+
+# Build and tag the auditor image.
+buildImage "auditor" "${audit_tags[@]}"
+# Push auditor images.
+pushImages "${audit_tags[@]}"
+
+if [ "$build_variant" == cip ]; then
+    # Define tags for the cip tool.
+    cip_tags=(
```

Comment:
This area looks really similar to the above; can we dedupe/simplify this?

Reply:
Ack.